dask/array/routines.py

```diff
@@ -515,34 +515,42 @@ def gradient(f, *varargs, **kwargs):
     return results
 
 
+def _bincount_sum(bincounts, dtype=int):
+    n = max(map(len, bincounts))
+    out = np.zeros(n, dtype=dtype)
+    for b in bincounts:
+        out[:len(b)] += b[:len(b)]
```

Comment:
```suggestion
        out[:len(b)] += b
```

Reply:
I recommend just pushing this in if you're working on this @jakirkham .  I tried committing from here, but it seems like I don't have rights to do this.